Bottom: e84588ff9b376b7b0be62cf7d4be0fde827308f4
Top:    fdd16824a1d23bca63d8371a89538e26f4616062
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-03 02:23:30 +0000

Refresh of play-with-perspective

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index afd0b86..ee92e0a 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1853,10 +1853,18 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
 
   glMatrixMode (GL_PROJECTION);
   glLoadIdentity ();
-  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+//  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+  glFrustum (-1, 1, -1, 1, 10, 10000);
   glMatrixMode (GL_MODELVIEW);
   glLoadIdentity ();
 
+  glScalef (2, 2, 1);
+  glTranslatef (-1., 1., 0.);
+  glScalef ( 2. / (float)widget->allocation.width,
+            -2. / (float)widget->allocation.height,
+             1. / 100.);
+  glTranslatef (0., 0., -2000.);
+
   glTranslatef (widget->allocation.width / 2., widget->allocation.height / 2., 0);
   glMultMatrixf ((GLfloat *)view_matrix);
   glTranslatef (-widget->allocation.width / 2., -widget->allocation.height / 2., 0);
@@ -2010,7 +2018,8 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
 
 
   // hid_expose_callback (&ghid_hid, &region, 0);
-  ghid_draw_everything (&region);
+  // ghid_draw_everything (&region);
+  ghid_draw_everything (NULL);
   hidgl_flush_triangles (&buffer);
 
   /* Just prod the drawing code so the current depth gets set to
